stripe/_api_requestor.py

```diff
@@ -99,36 +137,51 @@ def request(
         method: str,
         url: str,
         params: Optional[Mapping[str, Any]] = None,
-        headers: Optional[Mapping[str, str]] = None,
+        options: Optional[RequestOptions] = None,
         *,
+        base_address: BaseAddress,
+        api_mode: ApiMode,
         _usage: Optional[List[str]] = None,
-    ) -> Tuple[StripeResponse, str]:
-        rbody, rcode, rheaders, my_api_key = self.request_raw(
+    ) -> "StripeObject":
+        requestor = self._replace_options(options or {})
+        rbody, rcode, rheaders = requestor.request_raw(
             method.lower(),
             url,
             params,
-            headers,
             is_streaming=False,
+            api_mode=api_mode,
+            base_address=base_address,
+            options=options,
             _usage=_usage,
         )
-        resp = self.interpret_response(rbody, rcode, rheaders)
-        return resp, my_api_key
+        resp = requestor.interpret_response(rbody, rcode, rheaders, api_mode)
+
+        return _convert_to_stripe_object(
+            resp=resp,
+            params=params,
+            requestor=requestor,
+            api_mode=api_mode,
+        )
 
     def request_stream(
         self,
         method: str,
         url: str,
         params: Optional[Mapping[str, Any]] = None,
```

Comment:
I think we talked about it but why didn't we make params and options keyword only?

Reply:
In general, I only made new parameters keyword-only if they are required or have the potential to become required, but it would be disruptive to insert it into the middle of a list of positional parameters.